src/java/org/apache/cassandra/index/sai/utils/TypeUtil.java

```diff
@@ -278,7 +286,22 @@ else if (type instanceof DecimalType)
         return value;
     }
 
+    /**
+     * Inverse of {@link #encode}
```

Comment:
It might be worth a comment that mentions that integer and decimal types are not necessarily restored to their original precision.